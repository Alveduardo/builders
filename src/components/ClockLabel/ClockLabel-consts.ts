export const CLOCK_LABEL = {
  TEST_ID: {},
};
