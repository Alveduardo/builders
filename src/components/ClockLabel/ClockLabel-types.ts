import { ViewStyle } from 'react-native';

import { LabelProps } from '../Label';

interface StateConfig {
  date: string;
}

interface StylesConfig {
  containerStyle: ViewStyle;
}

export interface ClockLabelConfig {
  state: StateConfig;
  styles: StylesConfig;
}

export interface ClockLabelProps extends LabelProps {}
