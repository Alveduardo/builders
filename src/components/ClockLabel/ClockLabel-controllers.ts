import { useEffect, useState } from 'react';
import { ClockLabelConfig, ClockLabelProps } from './ClockLabel-types';

import useConstructor from '../../hooks/useConstructor';
import useInterval from '../../hooks/useInterval';
import useMounted from '../../hooks/useMounted';

import { getCurrentSeconds, getFormatedDate } from '../../utils/date/date-utils';
import { setupStyleProp } from '../../utils/style/style-utils';
import { ViewStyle } from 'react-native';

export const useClockLabelConfig = ({ style }: Partial<ClockLabelProps>): ClockLabelConfig => {
  const initialInterval = useConstructor(() => {
    return 60 - getCurrentSeconds();
  });

  const containerStyle = setupStyleProp(style) as ViewStyle;

  const mounted = useMounted();

  const [date, setDate] = useState(getFormatedDate());

  const intervalRef = useInterval(
    () => {
      setDate(getFormatedDate());
    },
    mounted.current ? 60000 : initialInterval,
  );

  useEffect(() => {
    return () => clearInterval(intervalRef.current);
  }, []);

  return {
    state: {
      date,
    },
    styles: {
      containerStyle,
    },
  };
};
