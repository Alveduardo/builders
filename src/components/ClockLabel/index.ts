import ClockLabel from './ClockLabel';
import { ClockLabelProps } from './ClockLabel-types';

export default ClockLabel;
export { ClockLabelProps };
