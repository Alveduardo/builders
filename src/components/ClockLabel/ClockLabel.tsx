import React, { memo } from 'react';

import { ClockLabelProps } from './ClockLabel-types';
import { useClockLabelConfig } from './ClockLabel-controllers';

import Label from '../Label';

const ClockLabel = memo(({ size, color, style, ...rest }: ClockLabelProps) => {
  const {
    state: { date },
    styles: { containerStyle },
  } = useClockLabelConfig({ style });

  return <Label {...{ rest, size, color, style: containerStyle }}>{date}</Label>;
});

export default ClockLabel;
