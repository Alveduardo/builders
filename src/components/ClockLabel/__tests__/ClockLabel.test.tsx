import React from 'react';
import { render } from '@testing-library/react-native';
import ClockLabel from '../ClockLabel';

describe('ClockLabel ->', () => {
  beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation(jest.fn());
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  afterAll(() => {
    jest.resetAllMocks();
  });

  describe('Snapshot ->', () => {
    it('Deve corresponder ao componente com as props default', () => {
      const tree = render(<ClockLabel />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('Deve corresponder ao componente com as props mais utilizadas', () => {
      const tree = render(<ClockLabel />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
