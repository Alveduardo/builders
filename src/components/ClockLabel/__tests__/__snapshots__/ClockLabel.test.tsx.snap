// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClockLabel -> Snapshot -> Deve corresponder ao componente com as props default 1`] = `
<Text
  rest={
    Object {
      "rest": Object {},
    }
  }
  style={
    Array [
      Object {
        "color": "#FFF",
        "fontSize": 16,
      },
      undefined,
    ]
  }
  testID="LABEL_CONTAINER_TEST_ID"
>
  21:34 - 3 Jul 2022
</Text>
`;

exports[`ClockLabel -> Snapshot -> Deve corresponder ao componente com as props mais utilizadas 1`] = `
<Text
  rest={
    Object {
      "rest": Object {},
    }
  }
  style={
    Array [
      Object {
        "color": "#FFF",
        "fontSize": 16,
      },
      undefined,
    ]
  }
  testID="LABEL_CONTAINER_TEST_ID"
>
  21:34 - 3 Jul 2022
</Text>
`;
